Add Price filter for French

rlustin opened the pull request:
@Signez, could you please review?

Signez commented on src/filters/price_filter.rs:
```diff
@@ -0,0 +1,79 @@
+
+    fn regex_pattern(&self) -> String {
+        match &self.locale as &str {
+            "fr" => format!("([\\d]+[^ ][{}]{{1}}|[{}]{{1}}[\\s]?[\\d]+)", self.currencies(), self.currencies()),
```
For future developers sanity, I think this line would benefit from a full-line comment above it, showing example of what it will produce (even parts of it).

rlustin replied:
Isn’t the purpose of tests in the same file?

Signez replied:
Not really; in my opinion, the provided tests help to understand the meaning of the filter, but not the "final" content of this regex. We could create tests for this individual function (that tests a fragment of it results, for exemple?) to help us understand what this regex looks like.

This is mainly the problem of complex regex: they are rarely ligitible. An English-written explanation could also helps a lot to dive into it, like `// Checks for digits followed by non-spaces, then by any of currencies() return value`… would be enough imho.

rlustin replied:
What’s about:
```rust
fn regex_pattern(&self) -> String {
    match &self.locale as &str {
        // Matches one of the following:
        // - digits followed by something other than a non-breaking space followed by any of
        //   currencies() return values (ex: `120€` or `120 $`);
        // - any of currencies() return values followed by any whitespace character (or not)
        //   followed by digits (ex: `€120` or `$ 120`).
        "fr" => format!("([\\d]+[^ ][{}]{{1}}|[{}]{{1}}[\\s]?[\\d]+)", self.currencies(), self.currencies()),
        _ => "".to_string(),
    }
}
```

Signez replied:
That would be really good. 👍

rlustin replied:
Fixed then.

Signez commented on src/filters/mod.rs:
```diff
@@ -42,5 +44,5 @@ pub trait LinterFilter {
 
     fn message(&self) -> &'static str;
-    fn regex_pattern(&self) -> &'static str;
+    fn regex_pattern(&self) -> String;
```
Shouldn't those changes be made in a separate commit?

rlustin replied:
You are right!

Signez reviewed (changes requested):
A few changes needed, but mainly okay. Good job!

rlustin committed:
Change regex_pattern() signature to output a String

Instead of a &'static str.

It allows us to construct patterns that are dynamic, and not static at
compile time.

rlustin committed:
Add PriceFilter for French
